components/arm/server.go

```diff
@@ -130,6 +133,55 @@ func (s *serviceServer) GetGeometries(ctx context.Context, req *commonpb.GetGeom
 	}
 	geometries, err := res.Geometries(ctx, req.Extra.AsMap())
 	if err != nil {
+		// if the error tells us the method is unimplemented, then we
+		// can use the kinematics and joint positions endpoints to
+		// construct the geometries of the arm
+		if strings.Contains(err.Error(), "unimplemented") {
+			var err error
+
+			kinematicsPbResp, err := s.GetKinematics(ctx, &commonpb.GetKinematicsRequest{Name: req.GetName()})
+			if err != nil {
+				return nil, err
+			}
+			format := kinematicsPbResp.GetFormat()
+			data := kinematicsPbResp.GetKinematicsData()
+			var model referenceframe.Model
+			switch format {
+			case commonpb.KinematicsFileFormat_KINEMATICS_FILE_FORMAT_SVA:
+				model, err = referenceframe.UnmarshalModelJSON(data, req.GetName())
+				if err != nil {
+					return nil, err
+				}
+			case commonpb.KinematicsFileFormat_KINEMATICS_FILE_FORMAT_URDF:
+				modelconf, err := urdf.UnmarshalModelXML(data, req.GetName())
+				if err != nil {
+					return nil, err
+				}
+				model, err = modelconf.ParseConfig(req.GetName())
+				if err != nil {
+					return nil, err
+				}
+			case commonpb.KinematicsFileFormat_KINEMATICS_FILE_FORMAT_UNSPECIFIED:
+				fallthrough
+			default:
+				if formatName, ok := commonpb.KinematicsFileFormat_name[int32(format)]; ok {
+					return nil, fmt.Errorf("unable to parse file of type %s", formatName)
+				}
+				return nil, fmt.Errorf("unable to parse unknown file type %d", format)
+			}
```

Comment:
All of this very nearly replicates the contents of `client.updateKinematics`

Let's make a helper function that does this.